### ISSUE TITLE
RoLS - regeneration

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#233]
```
Equip the RoLS, wait 5 seconds, then enter a new map.  As you enter the new map 
the Role's regeneration will fire and will cause the entire map to go black 
except for the tile your character is on.

What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?


What version of the product are you using? On what device?
0.6.9
DROID x 2.3


Please provide any additional information below.


```

Original issue reported on code.google.com by `jgemmajr@gmail.com` on 10 Aug 2011 at 2:10

**Comment by GoogleCodeExporter** [comment]
```
could reproduce this several times.
```

Original comment by `SamuelPl...@gmail.com` on 10 Aug 2011 at 6:24

**Comment by GoogleCodeExporter** [comment]
Original comment by `sdeva...@gmail.com` on 10 Aug 2011 at 6:21
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This is most likely caused by the double-buffering that is done by the Android 
framework. I'm considering introducing a "disable optimized drawing" setting 
that makes the game redraw the whole screen on every screenupdate. Currently, 
we only redraw the parts that have actually changed (which probably causes the 
problem).

I would like to solve this for v0.6.10.
```

Original comment by `oskar.wi...@gmail.com` on 13 Oct 2011 at 8:12
- Changed state: **Started**
- Added labels: **Milestone-0.6.10**

**Comment by GoogleCodeExporter** [comment]
```
What about invalidating some fields above / around the player.
```

Original comment by `SamuelPl...@gmail.com` on 14 Oct 2011 at 6:17

**Comment by GoogleCodeExporter** [comment]
```
r180 resolves parts of this, comfirmed by a user that exhibits the problem.
```

Original comment by `oskar.wi...@gmail.com` on 14 Oct 2011 at 6:31

**Comment by GoogleCodeExporter** [comment]
```
Will there be a downgrade of performance in older hardware disabling that?
```

Original comment by `sdeva...@gmail.com` on 15 Oct 2011 at 3:02

**Comment by GoogleCodeExporter** [comment]
```
re comment 6: On my device (HTC Desire), I did not notice any change in 
performance at all between having "Optimized drawing" enabled or disabled. If 
it works out ok on more devices, we might make even remove this option and 
always draw things the unoptimized way (redraw everything).

The default value for "Optimized drawing" is "disabled", which means that you 
have to opt-in to get the old (optimized) behaviour.
```

Original comment by `oskar.wi...@gmail.com` on 15 Oct 2011 at 7:28

**Comment by GoogleCodeExporter** [comment]
```
The fix in r180 has been confirmed by Nero on the forums. Marking this issue as 
ready for v0.6.10.
```

Original comment by `oskar.wi...@gmail.com` on 16 Oct 2011 at 5:19
- Changed state: **Ready**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.10 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 21 Oct 2011 at 8:56
- Changed state: **Fixed**
